wcfsetup/install/files/lib/core.functions.php

```diff
@@ -125,6 +125,56 @@ function getMinorVersion(): string {
 	use wcf\system\exception\SystemException;
 	use wcf\util\FileUtil;
 	use wcf\util\StringUtil;
+
+	/**
+	 * If the stacktrace contains a compiled template, the context of the relevant template line
+	 * is returned, otherwise an empty array is returned.
+	 */
+	function getTemplateContextLines(\Throwable $e): array {
+		try {
+			$contextLineCount = 5;
+			foreach ($e->getTrace() as $traceEntry) {
+				if (isset($traceEntry['file']) && \preg_match('~/templates/compiled/.+\.php$~',
+					$traceEntry['file'])) {
+					$startLine = $traceEntry['line'] - $contextLineCount;
+					if ($startLine < 0) {
+						$startLine = 0;
+					}
+					
+					$file = \fopen($traceEntry['file'], 'r');
+					if (!$file) {
+						break;
```

Comment:
All these `break`s could simply be a `return [];` after this became a dedicated function.